src/name_bazaar/ui/utils.cljs

```diff
@@ -152,14 +152,48 @@
       (str name ".eth"))))
 
 (defn try-resolving-address [db addr]
-  (if-not (web3/address? addr)
+  (when-not (web3/address? addr)
     (let [addr-patched (tldize addr)]
-      (if-let [resolved (some (fn [r]
-                                (get-in r [:public-resolver/records
-                                           (namehash addr-patched)
-                                           :public-resolver.record/addr]))
-                              (vals (:public-resolvers db)))]
-        resolved
-        "0x"))
-    addr))
-
+      (when-let [resolved (get-in db [:public-resolver/records
+                                      (namehash addr-patched)
+                                      :public-resolver.record/addr])]
+        resolved))))
+
+(defn try-reverse-resolving-address [db addr]
+  (when (web3/address? addr)
+    (when-let [resolved (get-in db [:public-resolver/reverse-records
+                                    addr
+                                    :public-resolver.record/name])]
+      resolved)))
+
+(defn resolve-params [db params]
+  (if (web3/address? (:user/address params))
+    (assoc params :user/ens-name (try-reverse-resolving-address db (:user/address params)))
+    (-> params
+        (assoc :user/ens-name (tldize (:user/address params)))
+        (assoc :user/address (try-resolving-address db (:user/address params))))))
+
+(defn human-address
+  ([params]
+   (human-address (:user/ens-name params)
+                  (:user/address params)))
```

Comment:
This is very chaotic. Util function should not make such assumptions about keys. Pls remove passing as map